gammapy/datasets/fermi.py

```diff
@@ -216,6 +218,135 @@ def fetch_fermi_diffuse_background_model(filename='gll_iem_v02.fit'):
 
     return filename
 
+class Fermi3FGLObject(object):
+    """TODO: Doccomments
+    """
+    fermi_cat = fetch_fermi_catalog('3FGL')
+
+    x_bins = np.log10([0.03, 0.1, 0.3, 1, 3, 10, 100])
+
+    y_labels = ['Flux30_100', 'Flux100_300', 'Flux300_1000',
+                'Flux1000_3000', 'Flux3000_10000', 'Flux10000_100000']
+
+    def __init__(self, source_name):
+        self.name_3FGL = source_name
+        self.catalog_index = np.where(self.fermi_cat[1].data['Source_Name'] == source_name)[0][0]
+        self.cat_row = self.fermi_cat[1].data[self.catalog_index]
+        self.ra = self.cat_row['RAJ2000']
+        self.dec = self.cat_row['DEJ2000']
+        self.gal_long = self.cat_row['GLON']
+        self.gal_lat = self.cat_row['GLAT']
+        self.int_flux = self.cat_row['Flux_Density']
+        self.unc_int_flux = self.cat_row['Unc_Flux_Density']
+        self.spec_type = self.cat_row['SpectrumType']
+        self.pivot_en = self.cat_row['PIVOT_ENERGY']
+        self.spec_index = self.cat_row['Spectral_Index']
+        self.unc_spec_index = self.cat_row['Unc_Spectral_Index']
+        self.beta = self.cat_row['beta']
+        self.unc_beta = self.cat_row['unc_beta']
+        self.cutoff = self.cat_row['Cutoff']
+        self.unc_cutoff = self.cat_row['Unc_Cutoff']
+        self.exp_index = self.cat_row['Exp_Index']
+        self.unc_exp_index = self.cat_row['Unc_Exp_Index']
+        self.signif = self.cat_row['Signif_Avg']
+
+    def plot_spectrum(self, ax=None):
+        import matplotlib.pyplot as plt
+
+        ax = plt.gca() if ax is None else ax
+
+
+
+        bin_edges1 = np.zeros(0)
+
+        bin_edges2 = np.zeros(0)
+
+        x_vals = np.zeros(0)
+
+        y_vals = np.zeros(0)
+
+        y_upper = np.zeros(0)
+        y_lower = np.zeros(0)
+
+
+        for i in range(0, np.size(self.x_bins) - 1):
```

Comment:
I'm sure this loop can be avoided by using clever numpy syntax or python list comprehension. You will find some suggestions in the notebook, that I've linked below.